File: clients_java/client/src/org/openamq/client/AMQTopic.java
/**
 * Date Created: 20-Jun-2005
 *************************************************************************
 * (c) Copyright JP Morgan Chase Ltd 2005. All rights reserved. No part of
 * this program may be photocopied reproduced or translated to another
 * program language without prior written consent of JP Morgan Chase Ltd
 *************************************************************************/
package org.openamq.client;

import javax.jms.JMSException;
import javax.jms.Topic;

/**
 * @author Robert Greig (robert.j.greig@jpmorgan.com)
 */
public class AMQTopic extends AMQDestination implements Topic
{
    public AMQTopic(String name)
    {
        this(name, false);
    }

    public AMQTopic(String name, boolean temporary)
    {
        this(name, DEFAULT_SCOPE, temporary);
    }

    public AMQTopic(String name, String scope, boolean temporary)
    {
        super(AMQDestination.TOPIC_EXCHANGE_NAME, AMQDestination.TOPIC_EXCHANGE_CLASS, name, temporary, null, scope);
    }

    public String getTopicName() throws JMSException
    {
        return super.getDestinationName();
    }

    public String getEncodedName()
    {
        return 'T' + getDestinationName();
    }

     public String getRoutingKey()
    {
        return getDestinationName();
    }
}

File: dotnet/xmsclient/Client/AMQTopic.cs
using System;
using jpmorgan.mina.common;
using log4net;


namespace OpenAMQ.XMS.Client
{
    public class AMQTopic : AMQDestination
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="AMQTopic"/> class.
        /// </summary>
        /// <param name="name">The name.</param>
        public AMQTopic(string name)
            : this(name, false)
        {            
        }

        /// <summary>
        /// Initializes a new instance of the <see cref="AMQTopic"/> class.
        /// </summary>
        /// <param name="name">The name.</param>
        /// <param name="temporary">if set to <c>true</c> [temporary].</param>
        public AMQTopic(string name, bool temporary)
            : base(AMQDestination.TOPIC_EXCHANGE_NAME, AMQDestination.TOPIC_EXCHANGE_CLASS, name, temporary, temporary, null)
        {            
        }
        
        public override string EncodedName
        {
            get
            {
                return 'T' + Name;
            }
        }

        public override string RoutingKey
        {
            get
            {
                return Name;
            }
        }
    }
}
